spec/features/trainees/withdraw_trainee_spec.rb

```diff
@@ -0,0 +1,189 @@
+      when_i_choose_another_day
+      and_enter_an_invalid_date
+      and_i_continue
+      then_i_see_the_error_message_for_invalid_date
+    end
+  end
+
+  context "trainee withdrawn for specific reason" do
+    background do
+      and_i_choose_a_specific_reason
+    end
+
+    scenario "today" do
+      when_i_choose_today
+      and_i_continue
+      then_i_am_redirected_to_withdrawal_confirmation_page
+      and_the_withdraw_date_and_reason_is_updated
+    end
+
+    scenario "yesterday" do
+      when_i_choose_yesterday
+      and_i_continue
+      then_i_am_redirected_to_withdrawal_confirmation_page
+      and_the_withdraw_date_and_reason_is_updated
+    end
+
+    scenario "on another day" do
+      when_i_choose_another_day
+      and_i_enter_a_valid_date
+      and_i_continue
+      then_i_am_redirected_to_withdrawal_confirmation_page
+      and_the_withdrawal_details_is_updated
+    end
+  end
+
+  context "trainee withdrawn for another reason" do
+    background do
+      and_i_choose_for_another_reason
+    end
+
+    scenario "and that reason is provided" do
+      when_i_choose_today
+      and_i_provide_a_reason
+      and_i_continue
+      then_i_am_redirected_to_withdrawal_confirmation_page
+      and_the_additional_reason_is_displayed
+    end
+
+    scenario "and a reason is not provided" do
+      and_i_continue
+      then_i_see_the_error_message_for_missing_additional_reason
+    end
+  end
+
+  def when_i_choose_today
+    when_i_choose("Today")
+  end
+
+  def when_i_choose_yesterday
+    when_i_choose("Yesterday")
+  end
+
+  def when_i_choose_another_day
+    when_i_choose("On another day")
+  end
+
+  def and_i_enter_a_valid_date
+    Faker::Date.in_date_period.tap do |withdraw_date|
+      withdrawal_page.set_date_fields(:withdraw_date, withdraw_date.strftime("%d/%m/%Y"))
+    end
+  end
+
+  def and_i_am_on_the_trainee_edit_page
+    edit_page.load(id: trainee.id)
+  end
+
+  def when_i_choose(option)
+    withdrawal_page.choose(option)
+  end
+
+  def and_i_continue
+    withdrawal_page.continue.click
+  end
+
+  def and_i_click_on_withdraw
+    edit_page.withdraw.click
+  end
+
+  def and_enter_an_invalid_date
+    withdrawal_page.set_date_fields(:withdraw_date, "32/01/2020")
+  end
+
+  def and_i_choose_a_specific_reason
+    label = I18n.t("views.forms.withdrawal_reasons.labels.#{WithdrawalReasons::SPECIFIC.sample}")
+    withdrawal_page.choose(label)
+  end
+
+  def and_i_choose_for_another_reason
+    label = I18n.t("views.forms.withdrawal_reasons.labels.#{WithdrawalReasons::FOR_ANOTHER_REASON}")
+    withdrawal_page.choose(label)
+  end
+
+  def and_i_provide_a_reason
+    withdrawal_page.additional_withdraw_reason.set(additional_withdraw_reason)
+  end
+
+  def then_i_see_the_error_message_for_invalid_date
+    expect(page).to have_content(
+      I18n.t("activemodel.errors.models.withdrawal_form.attributes.withdraw_date.invalid"),
+    )
+  end
+
+  def then_i_see_the_error_message_for_invalid_date
```

Comment:
I think this is the same method as the one above?

Reply:
Sharp eyes!